include/pagmo/serialization.hpp

```diff
@@ -29,13 +29,51 @@ see https://www.gnu.org/licenses/. */
 #ifndef PAGMO_SERIALIZATION_HPP
 #define PAGMO_SERIALIZATION_HPP
 
+// Let's disable a few compiler warnings emitted by the cereal code.
 #if defined(__clang__) || defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wconversion"
 #pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
 #pragma GCC diagnostic ignored "-Wdeprecated"
 #if __GNUC__ >= 7
 #pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+#endif
+#if defined(__clang__)
+
+#if defined(__apple_build_version__)
+
+// LLVM 3.2 -> Xcode 4.6.
+#if __clang_major__ > 4 || (__clang_major__ == 4 && __clang_minor__ >= 6)
+
+#pragma GCC diagnostic ignored "-Wunused-private-field"
+
+#endif
+
+// LLVM 3.7 -> Xcode 7.0.
+#if __clang_major__ >= 7
+
+#pragma GCC diagnostic ignored "-Wexceptions"
+
+#endif
+
+#else
+
+// LLVM 3.2.
+#if __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 2)
+
+#pragma GCC diagnostic ignored "-Wunused-private-field"
+
+#endif
+
+// LLVM 3.7.
+#if __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 7)
+
+#pragma GCC diagnostic ignored "-Wexceptions"
+
+#endif
+
+#endif
```

Comment:
a che serve sta roba?

Reply:
GO ahead and merge

Reply:
Selectively disable some clang warnings generated by the cereal headers.

It turns out that Apple, continuing their war against their users, hijack the clang version macro with Xcode's version string. So, let's say that you need to disable the "-Wexceptions" error warning, which is available since clang 3.7. You need to figure out from which version of Xcode clang 3.7 is being used internally, and write two sets of macros depending on whether you are using the "normal" clang or appleclang. Fun stuff.